arviz/stats/stats.py

```diff
@@ -865,6 +866,60 @@ def loo(data, pointwise=None, var_name=None, reff=None, scale=None):
         ],
     )
 
+def loo_expectation(data, values, pointwise=None, reff=None, **kwargs):
+    """
+    Computes the expectation of values with respect to the leave-one-out posteriors using PSIS.
+    Parameters
+    ----------
+        data: obj
+            Any object that can be converted to an :class:`arviz.InferenceData` object.
+            Refer to documentation of :func:`arviz.convert_to_dataset` for details.
+        values: ndarray
+            A vector of quantities to compute expectations for.
```

Comment:
We should add information about the shape of `values` here. My understanding is that is should be an array/dataarray with the same shape as the pointwise log likelihood (e.g. `chain, draw, obs_id`) which in general won't work here. I think there should be an extra check for when the input is a dataarray so that `chain, draw` dimensions get stacked into `__sample__` one, otherwise something like:

```
loo_expectation(data, data.posterior_predictive.y)
```

would not work as is.